$else $if not working with const

### Describe the bug

Using a $if ... $else $if ... $else block inside a const definition gives an error. However, leaving the $else $if portion out works. Note: The same happens when using an if ... else if ... else block.

### Expected Behavior

The following code:

```
import os

const bin =  $if linux { 
	os.join_path(@VMODROOT,'bin_linux_64') 
} $else $if macos { os.join_path(@VMODROOT,'bin_macos')  } $else { '' }


fn main() {
	println('Hello')
}
```

Should output 'Hello' both on Linux or Mac

### Current Behavior

But if gives:
```
$ v run .      
==================
/tmp/v_1000/haida.12639066475347183005.tmp.c:2731: error: declaration of void object
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

If you leave the $else $if out:
```
import os

const bin =  $if linux { 
	os.join_path(@VMODROOT,'bin_linux_64') 
}  $else { '' }


fn main() {
	println('Hello')
}
```

It runs:
```
$ v run .
Hello
```

### Reproduction Steps

Create a file .v with this content:
```
import os

const bin =  $if linux { 
	os.join_path(@VMODROOT,'bin_linux_64') 
} $else $if macos { os.join_path(@VMODROOT,'bin_macos')  } $else { '' }


fn main() {
	println('Hello')
}
```

Create two dummy files: 'bin_linux_x64' and 'bin_macos' in the same directory.

Run with v run .

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.0 36d45c6

### Environment details (OS name and version, etc.)

V full version: V 0.4.0 8b4c3fa.36d45c6
OS: linux, Debian GNU/Linux 11 (bullseye) (WSL 2)
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz

getwd: /home/rabt/devel
vexe: /home/rabt/devel/v/v
vexe mtime: 2023-07-21 13:39:23

vroot: OK, value: /home/rabt/devel/v
VMODULES: OK, value: /home/rabt/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.30.2
Git vroot status: weekly.2023.28-61-g36d45c6d
.git/config present: true

CC version: cc (Debian 10.2.1-6) 10.2.1 20210110
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1626,18 +1626,11 @@ fn (mut c Checker) const_decl(mut node ast.ConstDecl) {
 		}
 		node.fields[i].typ = ast.mktyp(typ)
 		if mut field.expr is ast.IfExpr {
-			if field.expr.branches.len == 2 {
-				first_stmts := field.expr.branches[0].stmts
-				second_stmts := field.expr.branches[1].stmts
-				if first_stmts.len > 0 && first_stmts.last() is ast.ExprStmt
-					&& first_stmts.last().typ != ast.void_type {
+			for branch in field.expr.branches {
+				if branch.stmts.len > 0 && branch.stmts.last() is ast.ExprStmt
+					&& branch.stmts.last().typ != ast.void_type {
 					field.expr.is_expr = true
-					field.expr.typ = (first_stmts.last() as ast.ExprStmt).typ
-					field.typ = field.expr.typ
-				} else if second_stmts.len > 0 && second_stmts.last() is ast.ExprStmt
-					&& second_stmts.last().typ != ast.void_type {
-					field.expr.is_expr = true
-					field.expr.typ = (second_stmts.last() as ast.ExprStmt).typ
+					field.expr.typ = (branch.stmts.last() as ast.ExprStmt).typ
 					field.typ = field.expr.typ
 				}
 			}

--- a/vlib/v/tests/const_from_multi_branchs_of_if_expr_test.v
+++ b/vlib/v/tests/const_from_multi_branchs_of_if_expr_test.v
@@ -1,0 +1,14 @@
+import os
+
+const bin = $if linux {
+	os.join_path(@VMODROOT, 'bin_linux_64')
+} $else $if macos {
+	os.join_path(@VMODROOT, 'bin_macos')
+} $else {
+	''
+}
+
+fn test_const_from_mutli_branchs_of_if_expr() {
+	println('Hello')
+	assert true
+}

